packages/ra-core/src/auth/useAuthState.ts

```diff
@@ -152,7 +152,7 @@ export type UseAuthStateResult<ErrorType = Error> = QueryObserverResult<
     boolean,
     ErrorType
 > & {
-    authenticated: QueryObserverResult<boolean, ErrorType>['data'];
+    authenticated?: QueryObserverResult<boolean, ErrorType>['data'];
```

Comment:
Wrong fix IMO, we must introduce the same types as the other hooks for each state

Reply:
Nice to have?